libcaf_net/caf/net/consumer.hpp

```diff
@@ -19,17 +19,22 @@
 #pragma once
 
 #include "caf/detail/net_export.hpp"
-#include "caf/make_counted.hpp"
-#include "caf/net/endpoint_manager.hpp"
-#include "caf/net/endpoint_manager_impl.hpp"
+#include "caf/net/consumer_queue.hpp"
 
 namespace caf::net {
 
-template <class Transport>
-endpoint_manager_ptr make_endpoint_manager(const multiplexer_ptr& mpx,
-                                           actor_system& sys, Transport trans) {
-  using impl = endpoint_manager_impl<Transport>;
-  return make_counted<impl>(mpx, sys, std::move(trans));
-}
+/// An implementation of BASP as an application layer protocol.
+class CAF_NET_EXPORT consumer {
```

Comment:
What is this good for and why do we need it? It seems to me like the proxies (they hold a pointer to this interface) could simply have a pointer to `basp::application` instead. Also avoiding dynamic dispatching entirely.

Reply:
The idea was to not include `basp` in the proxy. In hindsight this approach is not optimal. I can remove the interface and exchange the pointer for `basp::application*`.